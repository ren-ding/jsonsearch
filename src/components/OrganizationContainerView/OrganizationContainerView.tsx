import React, { Component } from 'react';
import OrganizationContainer from '../../model/OrganizationContainer';
import ItemView from '../ItemView';

export interface OrganizationContainerViewProps {
    organizationContainer: OrganizationContainer
}

export default class OrganizationContainerView extends Component<OrganizationContainerViewProps> {
    constructor(props:OrganizationContainerViewProps) {
      super(props);
    }
  
    render() {
      return (
        <div>
          <h2>Organization:</h2>
            <ItemView
              itemObject = {this.props.organizationContainer.organization}
            />
          <h2>Users:</h2>
            {this.props.organizationContainer.users.map(user => (
              <ItemView
                itemObject = {user}
              />
            ))}
          <h2>Tickets:</h2>
            {this.props.organizationContainer.tickets.map(ticket => (
              <ItemView
                itemObject = {ticket}
              />
            ))}
        </div>
      );
    }
}