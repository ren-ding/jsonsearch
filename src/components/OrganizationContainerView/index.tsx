import OrganizationContainerView from './OrganizationContainerView';
export default OrganizationContainerView;