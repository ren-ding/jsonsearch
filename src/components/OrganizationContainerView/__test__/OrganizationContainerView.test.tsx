import React from 'react';
import OrganizationContainerView, {OrganizationContainerViewProps} from '../OrganizationContainerView';
import renderer from 'react-test-renderer';

describe('OrganizationContainerView', () => {
    let props:OrganizationContainerViewProps;
    beforeEach(()=> {
        props = { 
            organizationContainer: {
                organization: {
                    "_id": 101,
                    "url": "http://initech.zendesk.com/api/v2/organizations/101.json",
                    "external_id": "9270ed79-35eb-4a38-a46f-35725197ea8d",
                    "name": "Enthaze",
                    "domain_names": [
                    "kage.com",
                    "ecratic.com",
                    "endipin.com",
                    "zentix.com"
                    ],
                    "created_at": "2016-05-21T11:10:28 -10:00",
                    "details": "MegaCorp",
                    "shared_tickets": false,
                    "tags": [
                    "Fulton",
                    "West",
                    "Rodriguez",
                    "Farley"
                    ]
                },
                users: [],
                tickets: []
            }
          }
    });

    it('should render to match snapshot', () => {
        const tree = renderer.create(<OrganizationContainerView {...props}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});