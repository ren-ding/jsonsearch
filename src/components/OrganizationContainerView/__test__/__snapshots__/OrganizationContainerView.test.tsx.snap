// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationContainerView should render to match snapshot 1`] = `
<div>
  <h2>
    Organization:
  </h2>
  <div>
    <table
      className="ui compact table"
    >
      <thead>
        <tr>
          <th>
            Field
          </th>
          <th>
            Value
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            _id
          </td>
          <td>
            101
          </td>
        </tr>
        <tr>
          <td>
            url
          </td>
          <td>
            http://initech.zendesk.com/api/v2/organizations/101.json
          </td>
        </tr>
        <tr>
          <td>
            external_id
          </td>
          <td>
            9270ed79-35eb-4a38-a46f-35725197ea8d
          </td>
        </tr>
        <tr>
          <td>
            name
          </td>
          <td>
            Enthaze
          </td>
        </tr>
        <tr>
          <td>
            domain_names
          </td>
          <td>
            kage.com,ecratic.com,endipin.com,zentix.com
          </td>
        </tr>
        <tr>
          <td>
            created_at
          </td>
          <td>
            2016-05-21T11:10:28 -10:00
          </td>
        </tr>
        <tr>
          <td>
            details
          </td>
          <td>
            MegaCorp
          </td>
        </tr>
        <tr>
          <td>
            shared_tickets
          </td>
          <td>
            false
          </td>
        </tr>
        <tr>
          <td>
            tags
          </td>
          <td>
            Fulton,West,Rodriguez,Farley
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <h2>
    Users:
  </h2>
  <h2>
    Tickets:
  </h2>
</div>
`;
